Plugins/ConnectSwiftPlugin/ConnectClientGenerator.swift

```diff
@@ -86,6 +91,32 @@ final class ConnectClientGenerator: Generator {
         self.printLine("}")
     }
 
+    private func printDescriptors(for service: ServiceDescriptor) {
+        let metadataTypeName = self.namer.typePrefix(forFile: service.file)
+            + NamingUtils.toUpperCamelCase(service.name)
+            + "Metadata"
```

Comment:
Would it be terrible to define this enum within the client type instead? It seems like that'd reduce the number of underscore-named top-level types and maybe be a bit more ergonomic.

(I'm a terrible Swift programmer, so this may be nonsense.)

Reply:
Yea I think that's a reasonable suggestion. I had originally avoided doing that since there's also an interface for the client (which wouldn't contain the metadata), so consumers would have to reference the concrete type. That said, the callout regarding reducing top-level names is a good enough reason to nest it in the concrete type